src/hotspot/share/opto/macro.cpp

```diff
@@ -2511,6 +2514,7 @@ bool PhaseMacroExpand::expand_macro_nodes() {
       }
       assert(!success || (C->macro_count() == (old_macro_count - 1)), "elimination must have deleted one node from macro list");
       progress = progress || success;
+      if (success) { C->print_method(PHASE_AFTER_MACRO_EXPANSION_ELIMINATE, 4, n); }
```

Comment:
I suggest to put it on a new line. 
```suggestion
      if (success) { 
        C->print_method(PHASE_AFTER_MACRO_EXPANSION_ELIMINATE, 4, n); 
      }
```
I think dumping at level 4 (same for the other newly added dumps) is too verbose already but when dumping at level 5, you have the overhead of additional IGVN step dumps. I could imagine that you either want macro expansion dumps OR IGVN step dumps but not always both. So, it looks like we've now reached a point where it would have been nice to enable/disable certain dumps. 

Anyway, since this is not possible I suggest to move these dumps to level 5 for now. But I'm open for other opinions about that.

Reply:
I think we should also add this to `eliminate_macro_nodes()` when eliminating locks, allocations, and boxing nodes. But then again, we also dump the graphs directly after EA since `eliminate_macro_nodes()` is also called from there. Not sure if that's wanted. I guess it's okay being at IGV dump level 5. But we could also restrict the dumps to the post loop opts phase.